File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/C/s196249385_comments_removed.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_N 1000

int main() {
    int n, k;
    scanf("%d %d", &n, &k);

    int a[MAX_N];
    for (int i = 0; i < n; i++) {
        scanf("%d", &a[i]);
    }

    long long cumsum[MAX_N + 1] = {0};
    for (int i = 0; i < n; i++) {
        cumsum[i + 1] = cumsum[i] + a[i];
    }

    long long b[MAX_N * (MAX_N + 1) / 2];
    int b_size = 0;
    long long max = 0;
    for (int i = 0; i < n; i++) {
        for (int j = i; j < n; j++) {
            b[b_size++] = cumsum[j + 1] - cumsum[i];
            if (b[b_size - 1] > max) {
                max = b[b_size - 1];
            }
        }
    }

    long long mask = 1;
    while (mask <= max) {
        mask <<= 1;
    }
    mask >>= 1;

    while (mask > 0) {
        long long tmp[MAX_N * (MAX_N + 1) / 2];
        int tmp_size = 0;
        for (int i = 0; i < b_size; i++) {
            if (b[i] & mask) {
                tmp[tmp_size++] = b[i];
            }
        }
        if (tmp_size >= k) {
            b_size = tmp_size;
            for (int i = 0; i < b_size; i++) {
                b[i] = tmp[i];
            }
        }
        mask >>= 1;
    }

    if (b_size < k) {
        printf("0\n");
    } else {
        long long ans = -1;
        for (int i = 0; i < b_size; i++) {
            ans &= b[i];
        }
        printf("%lld\n", ans);
    }

    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Go/s196249385_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	sc := bufio.NewScanner(os.Stdin)
	sc.Split(bufio.ScanWords)
	var n, k int
	if sc.Scan() {
		n, _ = strconv.Atoi(sc.Text())
	}
	if sc.Scan() {
		k, _ = strconv.Atoi(sc.Text())
	}

	a := make([]int, n)
	for i := 0; i < n; i++ {
		sc.Scan()
		num, _ := strconv.Atoi(sc.Text())
		a[i] = num
	}

	cumsum := make([]int64, n+1)
	for i := 0; i < n; i++ {
		cumsum[i+1] = cumsum[i] + int64(a[i])
	}

	var b []int64
	var max int64
	for i := 0; i < n; i++ {
		for j := i; j < n; j++ {
			sum := cumsum[j+1] - cumsum[i]
			b = append(b, sum)
			if sum > max {
				max = sum
			}
		}
	}

	mask := int64(1)
	for mask <= max {
		mask <<= 1
	}
	mask >>= 1

	for mask > 0 {
		tmp := make([]int64, 0)
		for _, e := range b {
			if (e & mask) != 0 {
				tmp = append(tmp, e)
			}
		}
		if len(tmp) >= k {
			b = tmp
		}
		mask >>= 1
	}

	if len(b) < k {
		fmt.Println(0)
	} else {
		ans := int64(-1)
		for _, e := range b {
			ans &= e
		}
		fmt.Println(ans)
	}
}

// End of Code
